src/globalsearch/spotifysearchprovider.cpp

```diff
@@ -90,30 +89,60 @@ void SpotifySearchProvider::SearchFinishedSlot(
   PendingState state = it.value();
   queries_.erase(it);
 
+  QMap<std::string, std::string> album_dedup;
   ResultList ret;
-  for (int i = 0; i < response.result_size(); ++i) {
-    const pb::spotify::Track& track = response.result(i);
-
-    Result result(this);
-    SpotifyService::SongFromProtobuf(track, &result.metadata_);
-
-    ret << result;
-  }
 
   for (int i = 0; i < response.album_size(); ++i) {
     const pb::spotify::Album& album = response.album(i);
 
+    QHash<QString, int> artist_count;
+    QString artist, majority_artist;
+    int majority_count = 0;
+
+    // Choose an album artist
+    for (int j = 0; j < album.track_size(); ++j) {
+      for (int k = 0; k < album.track(j).artist_size(); ++k) {
+        artist = QStringFromStdString(album.track(j).artist(k));
```

Comment:
... here, because you don't need it before, and don't reuse it outer the loop.
Usually, I prefer to keep the closest scope for object.
I don't think there is an overhead to do so, and even if there is one, it is probably very small.